pkg/kyverno/test/test_command.go

```diff
@@ -573,32 +595,181 @@ func getFullPath(paths []string, policyResourcePath string, isGit bool) []string
 }
 
 func applyPoliciesFromPath(fs billy.Filesystem, policyBytes []byte, valuesFile string, isGit bool, policyResourcePath string, rc *resultCounts) (err error) {
+	versionedvalues := &versionedTest{}
+
+	if err := json.Unmarshal(policyBytes, versionedvalues); err != nil {
+		return sanitizederror.NewWithError("failed to decode yaml", err)
+	}
+
+	// check for old format
+	if versionedvalues.APIVersion == "" {
+		err = nonVersionedPath(fs, policyBytes, valuesFile, isGit, policyResourcePath, rc)
+	} else {
+		err = versionedPath(fs, policyBytes, valuesFile, isGit, policyResourcePath, rc)
+	}
+
+	return err
+
+}
+
+func versionedPath(fs billy.Filesystem, policyBytes []byte, valuesFile string, isGit bool, policyResourcePath string, rc *resultCounts) (err error) {
 	openAPIController, err := openapi.NewOpenAPIController()
+	if err != nil {
+		return sanitizederror.NewWithError("failed to get new openAPIcontroller", err)
+	}
 	engineResponses := make([]*response.EngineResponse, 0)
 	var dClient *client.Client
-	values := &Test{}
+	versionedvalues := &versionedTest{}
 	var variablesString string
 	var pvInfos []policyreport.Info
 	var resultCounts common.ResultCounts
+	var resultsMap map[string]report.PolicyReportResult
+	var testResults []TestResults
+	var policyFullPath []string
+	var resourceFullPath []string
 	store.SetMock(true)
 
-	if err := json.Unmarshal(policyBytes, values); err != nil {
+	if err := json.Unmarshal(policyBytes, versionedvalues); err != nil {
 		return sanitizederror.NewWithError("failed to decode yaml", err)
 	}
 
+	fmt.Printf("\nExecuting %s...", versionedvalues.MetaData.Name)
+	log.Log.V(5).Info("valuesFile = ", valuesFile)
+	valuesFile = versionedvalues.Spec.Variables
+
+	variables, globalValMap, valuesMap, namespaceSelectorMap, err := common.GetVariable(variablesString, valuesFile, fs, isGit, policyResourcePath)
+	if err != nil {
+		if !sanitizederror.IsErrorSanitized(err) {
+			return sanitizederror.NewWithError("failed to decode yaml", err)
+		}
+		return err
+	}
+
+	policyFullPath = getFullPath(versionedvalues.Spec.Policies, policyResourcePath, isGit)
```

Comment:
```suggestion
	policyFullPath := getFullPath(versionedvalues.Spec.Policies, policyResourcePath, isGit)
```

Reply:
I now understand what you meant. I will do this and push the changes.
Thank you very much!

Reply:
I have tried to reflect this in my latest commit, please let me know if it is okay
Thank you!